### PR TITLE
Add plugin xmind

### PR DESCRIPTION
**Pull request by sudoforge**
Replacing #532; the png image was not originally added with `git-lfs`, so rebasing was necessary.

### CODE REVIEW

**Comment by danstewart** [comment]
@bddenhartog @89luca89 Having a few issues when testing this.

1. The URL 404s (both in the browser and via `wget`).
The website download button links to http://www.xmind.net/xmind/downloads/xmind-8-update7-linux.zip - which works in the browser but 403s using `wget`...
Looks like that URL redirects to https://dl3.xmind.net/xmind-8-update7-linux.zip - which does work from `wget`.

2. The `.desktop` file `exec` is a directory. It should link to `/opt/xmind/XMind_${cpuArchTag}/XMind`

3. The architecture check is incorrect, should be `[[ "$cpuArch" =~ "64" ]] && cpuArchTag="amd64" || cpuArchTag="i386"`

4. More Java errors. The ini file has relative paths in it, so it fails to start from `/usr/bin`
See: https://groups.google.com/d/msg/xmind-dev/VhjWlOL6dDE/dvO9f5OknhwJ
This worked fine once I `chown`-ed `/opt/xmind` and ran without `sudo` from the `XMind_${cpuArchTag}` directory.
We'll either have to rewrite the `ini` file or install it to the users home dir.

File: plugins/xmind.plugin/install.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+CACHEDIR="/var/cache/fedy/xmind";
+
+mkdir -p "$CACHEDIR"
+cd "$CACHEDIR"
+
+URL="http://dl2.xmind.net/xmind-downloads/xmind-8-update3-linux.zip"
```
Comment (danstewart) [inline plugins/xmind.plugin/install.sh]: Should be http://www.xmind.net/xmind/downloads/xmind-8-update7-linux.zip

File: plugins/xmind.plugin/install.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+if [[ ! -f "xmind.zip" ]]; then
+	exit 1
+fi
+rm -rf "/opt/xmind"
+mkdir -p "/opt/xmind"
+unzip -xq "xmind.zip" -d "/opt/xmind"
+
+cpuArch=$(lscpu | grep 'Architecture' | awk '{print $2}')
+[ "$cpuArch" = "64" ] && cpuArchTag="amd64" || cpuArchTag="i386"
+ln -sf "/opt/xmind/XMind_${cpuArchTag}" "/usr/bin/xmind"
```
Comment (danstewart) [inline plugins/xmind.plugin/install.sh]: Should link to `/opt/xmind/XMind_${cpuArchTag}/XMind`

File: plugins/xmind.plugin/install.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+if [[ ! -f "xmind.zip" ]]; then
+	exit 1
+fi
+rm -rf "/opt/xmind"
+mkdir -p "/opt/xmind"
+unzip -xq "xmind.zip" -d "/opt/xmind"
+
+cpuArch=$(lscpu | grep 'Architecture' | awk '{print $2}')
+[ "$cpuArch" = "64" ] && cpuArchTag="amd64" || cpuArchTag="i386"
```
Comment (danstewart) [inline plugins/xmind.plugin/install.sh]: Should be - `[[ "$cpuArch" =~ "64" ]] && cpuArchTag="amd64" || cpuArchTag="i386"`
Doesn't work for `x86_64`

File: plugins/xmind.plugin/install.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ln -sf "/opt/xmind/XMind_${cpuArchTag}" "/usr/bin/xmind"
+
+wget -c "http://s3.amazonaws.com/xmindnet/blog/en/new_branding_new_logo" -O xmind.png
+xdg-icon-resource install --novendor --size "512" "/opt/xmind/resources/app.asar.unpacked/assets/xmind.png" "xmind"
+gtk-update-icon-cache -f -t /usr/share/icons/hicolor
+
+cat <<EOF | tee /usr/share/applications/xmind.desktop
```
Comment (danstewart) [inline plugins/xmind.plugin/install.sh]: The `ini` file uses relative paths so fails to run from home directory.

**Comment by bddenhartog** [comment]
@danstewart thanks for looking over it, i hadn't tested it -- just fixed the LFS issue and pushed it. I'll tackle the issues here (and in other PRs), I'm pretty sure @89luca89 has abandoned the PRs he has here.

**Pull request closed without merging**